cmd/cca/check_test.go

```diff
@@ -192,3 +192,26 @@ func Test_CheckCmd_token_invalid_format(t *testing.T) {
 	err = cmd.Execute()
 	assert.EqualError(t, err, expectedErr)
 }
+
+func Test_CheckCmd_regression_bug_18(t *testing.T) {
+	fs := afero.NewMemMapFs()
+
+	err := afero.WriteFile(fs, "ccatoken.cbor", testValidCCAToken, 0644)
+	require.NoError(t, err)
+
+	err = afero.WriteFile(fs, "not-a-public-key.jwk", testValidPAK, 0644)
+	require.NoError(t, err)
+
+	cmd := NewCheckCmd(fs)
+	cmd.SetArgs(
+		[]string{
+			"--token=ccatoken.cbor",
+			"--key=not-a-public-key.jwk",
+		},
+	)
+
+	expectedErr := `verifying CCA evidence from ccatoken.cbor using key from not-a-public-key.jwk: unable to verify platform token: unable to instantiate verifier: ES256: algorithm mismatch`
```

Comment:
this error does not seem to indicate the type of test which is intended ?

The error says algorithm mismatch however as I understand is that it is not a public key but a provate key used to verify the signature on the token?

Reply:
Yeah, I had the same reaction.  The last bit of the error string is completely misleading.  If it ended with "unable to instantiate verifier" it'd be awesome.  Unfortunately, go-cose wants to be helpful (see https://github.com/veraison/go-cose/blob/07c7d28e2c7eb9d6daa4619b98eceeaf7a8e5156/verifier.go#L47) but misinterprets the situation.

Reply:
Issue https://github.com/veraison/go-cose/issues/124 raised on go-cose to track this!